### PR TITLE
DOC: Document Remote Code Execution risk for Dataframe.query and computation.eval

### PR DESCRIPTION
**Pull request by r0rshark**
## TLDR
This diff adds documentation to the Remote Code Execution risk for `Dataframe.query` and `computation.eval` with the same wording used in [Dataframe.eval](https://github.com/pandas-dev/pandas/blob/v2.2.2/pandas/core/frame.py#L4850-L4852)

## Details
The Dataframe.query function is internally calling Dataframe.eval [here](https://github.com/pandas-dev/pandas/blob/b552dc95c9fa50e9ca2a0c9f9cdb8757f794fedb/pandas/core/frame.py#L4547) passing the `expr` parameter. 
Dataframe.eval is correctly warning in the [documentation](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.eval.html#pandas.DataFrame.eval) that this could lead to Code execution risk.
This is possible because using the `@` syntax we can access local variables and by referencing the imported pandas library we can get the `os.system` function thanks to `os` being imported inside [compat/__init__.py](https://github.com/pandas-dev/pandas/blob/main/pandas/compat/__init__.py#L13). In this way passing something like `@pd.compat.os.system('whoami')` to the `expr` parameter of `Dataframe.query` we can achieve code execution.

However in Dataframe.query [documentation](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.query.html) we do not surface this risk we only point out that pandas.eval is called but also [there](https://pandas.pydata.org/docs/reference/api/pandas.eval.html#pandas.eval) we do not warn about any code execution risks. 

The vulnerable code I found and that also surprised the developer who wrote it of the code execution risk can be summarized in UserControlled input flowing into the DataFrame.query expr value similarly to
```
import pandas as pd

def api_whatever(request: HTTPRequest):
  filter = request.GET["filter"]
  data = {
    "test": [125, 230, 412],
  }
  df = pd.DataFrame(data)
  df.query(filter)
   ... 

```
Calling this api with data:
https://www.example.com?filter=@pd.compat.os.system('whoami')
will allow code execution.

### CODE REVIEW

**Review by WillAyd — APPROVED** [review]
lgtm @jorisvandenbossche

File: pandas/core/frame.py [hunk 1]
```diff
@@ -4473,6 +4473,10 @@ def query(self, expr: str, *, inplace: bool = False, **kwargs) -> DataFrame | No
         """
         Query the columns of a DataFrame with a boolean expression.
 
+        This function pass the `expr` parameter to :meth:`~pandas.DataFrame.eval`.
+        This allows `eval` to run arbitrary code, which can make you vulnerable to code
+        injection if you pass user input to this function.
```
Comment (rhshadrach) [inline pandas/core/frame.py:4478]:
```suggestion
        This method can run arbitrary code, which can make you vulnerable to code
        injection if you pass user input to this function.
```

I don't think we need to mention the implementation details; it's sufficient to warn that query can run arbitrary code.

File: pandas/core/computation/eval.py [hunk 2]
```diff
@@ -193,6 +193,8 @@ def eval(
     corresponding bitwise operators.  :class:`~pandas.Series` and
     :class:`~pandas.DataFrame` objects are supported and behave as they would
     with plain ol' Python evaluation.
+    This allows `eval` to run arbitrary code, which can make you vulnerable to code
+    injection if you pass user input to this function.
```
Comment (rhshadrach) [inline pandas/core/computation/eval.py:197]: This is an internal method, right? In such case, I don't think this is necessary.

**Review by rhshadrach — CHANGES_REQUESTED** [review]
Thanks for the PR!

File: pandas/core/computation/eval.py (hunk 2 above)
Comment (rhshadrach) [reply pandas/core/computation/eval.py:197]: Ah - this is exposed at the top level. Agree with including it, but I don't think starting with `This` is accurate - it 
 reads to me like that refers to the previous sentence. Can just say

> `eval` can run arbitrary code, ...

**Review by rhshadrach — APPROVED** [review]
lgtm

**Comment by rhshadrach** [comment]
Thanks @r0rshark!
